gstd/gstd_daemon.c

```diff
@@ -72,6 +87,32 @@ gstd_daemon_init (gint argc, gchar * argv[], gchar * pidfilename)
   daemon_pid_file_proc = gstd_daemon_pid;
 
   _initialized = TRUE;
+  goto out;
+
+error:
+  {
+    switch (errno) {
+      case EACCES:
+        g_printerr ("User %s must have write permissions to %s and its contents\n",
+            g_get_user_name (), pid_path);
+        break;
+      case ENOENT:
+        g_printerr ("Directory %s does not exist, please create it and grant write "
+            "permissions to user %s\n", pid_path, g_get_user_name ());
+        break;
+      default:
+        g_printerr ("Failed to access %s with user %s\n", pid_path, g_get_user_name ());
+        break;
+    }
+  }
+
+out:
```

Comment:
This needs to be refactored. The only `goto` usage I recommend is the `single-input / single-output` pattern and it is used as a clean mechanism to handle de-initialization/deallocation. In this case, it is not used for this and hence it looks messy. I recommend the following:

```c
gst_daemon_init (gint argc, gchar * argv[], gchar * pidfilename)
{
  const gchar *process_name;
  gchar *pid_path = NULL;
  return ret = TRUE;

  g_return_val_if_fail (argv, FALSE);

  if (_initialized)
    goto out;

  /* Check if pid dir available */
  pid_path = gstd_daemon_get_pid_filename (pidfilename);
  if (NULL == pid_path) {
    g_printerr ("Unable to access Gstd PID dir: path is NULL");
    ret = FALSE;
    goto freepath;
  }

  if (g_access (pid_path, W_OK)) {
    /* If you think the switch is needed, I'd put it here.
     */
    g_printerr ("Unable to access Gstd PID dir: %s\n", strerror(errno));
    ret = FALSE;
    goto freepath;
  }

  /* Sanitize the process name to use it as PID identification */
  process_name = daemon_ident_from_argv0 (argv[0]);
  
  /* Use the process name as the identification prefix for the 
     pid file */
  daemon_pid_file_ident = process_name;
 
   /*
     Create a gstd.pid file in another path, it could be 
     nullable  */
  pid_filename = pidfilename;

  /* Override the default pid file location to /tmp/ to avoid the need
     of root privileges */
  daemon_pid_file_proc = gstd_daemon_pid;

  _initialized = TRUE;

freepath:
  free(pid_path);
out:
  return ret;
}
```

Reply:
Fixed